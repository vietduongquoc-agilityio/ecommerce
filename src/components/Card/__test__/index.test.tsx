import { render, screen } from "@testing-library/react";
import { useRouter } from "next/navigation";
import ItemCard from "..";

jest.mock("next/navigation", () => ({
  useRouter: jest.fn(),
}));

describe("ItemCard Component", () => {
  const mockRouter = { push: jest.fn() };
  (useRouter as jest.Mock).mockReturnValue(mockRouter);

  const mockItem = {
    name: "Test Product",
    description: "This is a test product",
    price: 100000,
    image: "/test-image.jpg",
  };

  it("should render correctly and match snapshot", () => {
    const { asFragment } = render(<ItemCard id={0} {...mockItem} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("should display correct product details", () => {
    render(<ItemCard id={0} {...mockItem} />);

    expect(screen.getByText("Test Product")).toBeInTheDocument();
    expect(screen.getByText("This is a test product")).toBeInTheDocument();
    expect(screen.getByText("Rp 100.000")).toBeInTheDocument();
  });
});
