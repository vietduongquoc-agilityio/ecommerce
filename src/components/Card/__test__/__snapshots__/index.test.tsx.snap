// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemCard Component should render correctly and match snapshot 1`] = `
<DocumentFragment>
  <article
    style="position: relative; width: 285px; height: 446px; background: rgb(244, 245, 247); overflow: hidden; cursor: pointer; margin-bottom: 32px;"
  >
    <img
      alt="Test Product"
      data-nimg="1"
      decoding="async"
      height="300"
      loading="lazy"
      src="/_next/image?url=%2Ftest-image.jpg&w=640&q=75"
      srcset="/_next/image?url=%2Ftest-image.jpg&w=384&q=75 1x, /_next/image?url=%2Ftest-image.jpg&w=640&q=75 2x"
      style="color: transparent; object-fit: cover;"
      width="285"
    />
    <div
      style="padding: 16px;"
    >
      <p
        style="color: rgb(58, 58, 58); font-weight: 600; font-size: 24px; margin-bottom: 8px;"
      >
        Test Product
      </p>
      <p
        style="color: rgb(159, 159, 159); font-weight: 500; font-size: 16px; margin-bottom: 8px;"
      >
        This is a test product
      </p>
      <p
        style="color: rgb(58, 58, 58); font-weight: 600; font-size: 20px;"
      >
        Rp 100.000
      </p>
    </div>
    <div
      style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; background-color: rgb(58, 58, 58); opacity: 0; display: flex; align-items: center; justify-content: center; transition: opacity 0.3s ease; pointer-events: none;"
    >
      <button
        style="padding: 12px 52px; font-size: 18px; font-weight: 600; text-align: center; width: auto; height: auto; cursor: pointer; transition: all 0.3s ease; background-color: rgb(255, 255, 255); color: rgb(184, 142, 47);"
      >
        Add to cart
      </button>
    </div>
  </article>
</DocumentFragment>
`;
